Split Resources Tab

KevWho opened the pull request:
Splits the Campus-Wide tab of the Resources tab into three distinct tags: `Basic Needs`, `Health`, `Admin` (in that order). Would like feedback about this change:
- Ordering of the categories.
- Would we want to not hardcode these categories in the case that new ones are added to the backend, or is that not a concern and having deliberate ordering is preferable?
- How does the scrolling of the segmented control feel / is it intuitive, or will many users not know that there may be obscured options? On the iPhone 12 Pro, `Admin` is completely cut off, and I'm not sure how to indicate to users that there are more pages.

## Description

Since the additional buttons flow off the edge of the screen for small screens, users can scroll through the options of the `SegmentedControl` horizontally.
- Adds the public `sizeEqually` and `spacing` variables to `SegmentedControl`.
  - When `sizeEqually` is `false`, the control will size the widths of the segments based on their text lengths, rather than having equal-sized segments that shrink the font size of longer labels when there is no room (previous behavior).
  - The `spacing` value is only used when `sizeEqually` is `false`, and determines the margins between segments.
- If the user swipes to a page whose label is cut-off in the segmented control, automatically scroll the segmented control to display the full text.

Splits the Resources tab into `Basic Needs`, `Health`, and `Admin`.
- Removes the `.finances` and `.legal` types from the `ResourceType` enum.
- Changes `CampusResourceViewController` to display only the filtered resources of a single type.

## After

<img width="559" alt="Screen Shot 2021-01-25 at 10 07 15 PM" src="https://user-images.githubusercontent.com/41145903/105806908-b5196300-5f59-11eb-9d1c-ebeefc79d1a9.png">

huangshawn1 commented on berkeley-mobile/Resources/ResourcesDataSource/ResourceType.swift:
```diff
@@ -12,19 +12,13 @@ import UIKit
 
     case health = "Health"
-    case finances = "Finances"
-    case legal = "Legal"
     case basicNeeds = "Basic Needs"
     case admin = "Admin"
```
I'd consider adding an unknown/other category in case of future changes. that way, everything backend has (including any new resources) will show up even if not categorized correctly.

huangshawn1 reviewed (commented):
personally, I would put basic needs last and maybe do (Health, Admin, Basic Needs) based on what I think the majority of students would use the most.
the cutoff for the scrolling categories seems a bit too far to the left, it looks a bit awkward to me. can't think of any better way to indicate scrollable other than having one of the labels be cut off.
![IMG_1922](https://user-images.githubusercontent.com/20230668/105809049-d2e8c700-5f5d-11eb-9c19-f7aad6483cd2.PNG)

KevWho commented:
> personally, I would put basic needs last and maybe do (Health, Admin, Basic Needs) based on what I think the majority of students would use the most.

I based the ordering off of this, but I can change it around.

<img width="497" alt="Screen Shot 2021-01-25 at 10 42 51 PM" src="https://user-images.githubusercontent.com/41145903/105809942-f19b8d80-5f5e-11eb-9d1a-7643af7f346a.png">

Stereopoly commented:
I think some sort of fading action on the left and right edges of the segmented control could help show that it's movable

KevWho commented:
Pushed some changes: added a right inset to the controls, reorganized the tabs to `Health`, `Admin`, `Basic Needs`, and added an `Other` tab.

Also added a button to the right of the controls that is displayed when the user is not on the last tab and when the user is not scrolled all the way to the right:
<img width="560" alt="Screen Shot 2021-01-30 at 3 32 22 PM" src="https://user-images.githubusercontent.com/41145903/106370621-8bbd5600-6310-11eb-8346-f3a369a1acc4.png">

huangshawn1 reviewed (approved):
looks good, shadow looks a bit strange on the arrow button

KevWho commented:
w/o shadow:
<img width="560" alt="Screen Shot 2021-01-30 at 3 46 55 PM" src="https://user-images.githubusercontent.com/41145903/106370802-747f6800-6312-11eb-9853-12ef31253fd5.png">

KevWho commented:
w/ same shadow as `CardView`:
<img width="560" alt="Screen Shot 2021-01-30 at 3 54 15 PM" src="https://user-images.githubusercontent.com/41145903/106370890-7dbd0480-6313-11eb-816c-e6fcb0489735.png">
